pkg/recipes/terraform/execute.go

```diff
@@ -200,13 +205,14 @@ func (e *executor) GetRecipeMetadata(ctx context.Context, options Options) (map[
 	}, nil
 }
 
-// setEnvironmentVariables sets environment variables for the Terraform process by reading values from the environment configuration.
+// setEnvironmentVariables sets environment variables for the Terraform process by reading values from the recipe configuration.
 // Terraform process will use environment variables as input for the recipe deployment.
-func (e executor) setEnvironmentVariables(ctx context.Context, tf *tfexec.Terraform, envConfig *recipes.Configuration) error {
-	if envConfig != nil && envConfig.RecipeConfig.Env.AdditionalProperties != nil {
-		envVars := map[string]string{}
+func (e executor) setEnvironmentVariables(ctx context.Context, tf *tfexec.Terraform, recipeConfig *datamodel.RecipeConfigProperties) error {
+	if recipeConfig != nil && recipeConfig.Env.AdditionalProperties != nil && len(recipeConfig.Env.AdditionalProperties) > 0 {
+		// populate envVars with the environment variables from current process
+		envVars := splitEnvVar(os.Environ())
```

Comment:
So, we first need to create a map of existing environment variables (**os.Environ()**) and then add our environment variables to the map and then set them again.

Can't we just create a map of environment variables to be added and call `tf.SetEnv`? Not sure why we need to get all the environment variables (existing ones) and re-set them?

Reply:
tf.SetEnv() does not copy over the environment variables from current process, which caused errors. we discussed and have updated code to include them.